modules/text_server_adv/text_server_adv.cpp

```diff
@@ -4526,6 +4526,12 @@ bool TextServerAdvanced::_shape_substr(ShapedTextDataAdvanced *p_new_sd, const S
 					if ((sd_glyphs[j].start >= bidi_run_start) && (sd_glyphs[j].end <= bidi_run_end)) {
 						// Copy glyphs.
 						Glyph gl = sd_glyphs[j];
+						if (gl.end == p_start + p_length && ((gl.flags & GRAPHEME_IS_SHY) == GRAPHEME_IS_SHY)) {
+							uint32_t index = font_get_glyph_index(gl.font_rid, gl.font_size, 0x00ad, 0);
+							float w = (p_new_sd->orientation == ORIENTATION_HORIZONTAL) ? font_get_glyph_advance(gl.font_rid, gl.font_size, index).x : font_get_glyph_advance(gl.font_rid, gl.font_size, index).y;
```

Comment:
```suggestion
							float w = 
 font_get_glyph_advance(gl.font_rid, gl.font_size, index)[(p_new_sd->orientation == ORIENTATION_HORIZONTAL) ? 0 : 1];
```
Alternative, compact approach, also more readable IMO as you don't have to ensure both sides of the ternary are the same